### ISSUE TITLE
Optuna GirdSampler error

### ISSUE THREAD

**Issue by yfeng24816** [facebookresearch/hydra#2003]
# 🐛 Bug
## Description
Got this error when using Optuna random sampler
<img width="1253" alt="image" src="https://user-images.githubusercontent.com/83372333/152044607-64e13548-fc9a-41a6-acfe-13d3aaccd84e.png">

This is my config
<img width="653" alt="image" src="https://user-images.githubusercontent.com/83372333/152044659-6e7d21c6-52b8-4372-b476-63db46f4dd32.png">

This is weird because `consider_prior` is not in the config of RandomSampler of Hydra
<img width="811" alt="image" src="https://user-images.githubusercontent.com/83372333/152046000-4ab9fba6-d174-4923-9440-d3081535669c.png">

## Update(Debugging)
This seems to be hydra defaults to load the TPE config. 
<img width="949" alt="image" src="https://user-images.githubusercontent.com/83372333/152049096-641d07ff-2e62-4f19-b75d-0fe803cc567c.png">

In this file, it first loads the defaulted TPE config, then my randomSampler config. But when doing the merge, the randomSampler's config did not override the redundant parts like `consider_prior`, making them passed to Optuna's RandomSampler, and caused error. 
<img width="272" alt="image" src="https://user-images.githubusercontent.com/83372333/152049347-0095e58b-251e-4ac6-a969-cf48c0b5ad54.png">
<img width="1283" alt="image" src="https://user-images.githubusercontent.com/83372333/152049371-76a0097d-600d-494d-879d-df771a3c35d7.png">

<img width="1553" alt="image" src="https://user-images.githubusercontent.com/83372333/152049288-a56c0447-8506-4180-8ee0-3ea07259fce8.png">

Is there a way to manipulated the defaulted TPE config and avoid the error?

## Checklist
- [x] I checked on the latest version of Hydra
- [ ] I created a minimal repro (See [this](https://stackoverflow.com/help/minimal-reproducible-example) for tips).

**Comment by Jasha10** [comment]
Hi @yfeng24816,
Yes, Hydra selects the TPE sampler by default (see [here](https://github.com/facebookresearch/hydra/blob/8d45045e1408ca9d67a504c2372fb732bb9e6920/plugins/hydra_optuna_sweeper/hydra_plugins/hydra_optuna_sweeper/config.py#L129)).
You need to override the `hydra/sweeper/sampler` group using the defaults list.
Please try adding the following to your config:
```yaml
defaults:
  - /hydra/sweeper/sampler: random  # using RandomSamplerConfig
```

**Comment by Jasha10** [comment]
@yfeng24816 did this solve your problem?

**Comment by yfeng24816** [comment]
> @yfeng24816 did this solve your problem?

Yes Jasha, thank you. Closing this issue.

**Comment by yfeng24816** [comment]
Is there a way to use Optuna's GridSampler? I tried this in config but failed

<img width="355" alt="image" src="https://user-images.githubusercontent.com/83372333/159147786-f59193fd-d5f4-4eba-a2f7-77c33bf17f49.png">
<img width="587" alt="image" src="https://user-images.githubusercontent.com/83372333/159147793-b4a3f953-f612-4e78-93de-00324d1d23b9.png">

It is possible to use hydra's -m option with command line interface for grid search, and in fact I used it pretty often. But I just hope if there is something more consistent. And Optuna's hyper parameter search outputs a `optimization_results.yaml` which I found very handy for loading search result.
